### ISSUE TITLE
Field injection for selective logging build

### ISSUE THREAD

**Issue by kartikeytewari-ul** [unloggedio/unlogged-sdk#98]
### Describe the bug
Injected field in selective logging, causes issues when reflection is used on the class.

### Reproduction steps
1. Compile a class with unlogged-sdk.
2. Use reflection on the class to get it's fields.
3. Unlogged injected hash map would also come up in the class.

### Expected behaviour
No such injected field should have come up in the class as this can cause issues downstream in user code.

### Example
1. Java Code
```java
package org.unlogged.demo.controller;

public class testClass {

    public int getAdd(int a, int b) {
        return a+b;
    }
}
```

2. Decompiled Class now
```java
//
// Source code recreated from a .class file by IntelliJ IDEA
// (powered by FernFlower decompiler)
//

package org.unlogged.demo.controller;

import io.unlogged.Runtime;
import io.unlogged.logging.Logging;
import java.util.HashMap;

public class testClass {
    static HashMap org$unlogged$demo$controller$testClass$unlogged$$mapStore;

    public testClass() {
        int var1 = 0;
        Logging.recordEvent(701676);
        Logging.recordEvent(701677);
        Logging.recordEvent(701678);
        var1 = 701678;
        super();
        byte var3 = 0;
        Logging.recordEvent(701679);
        Logging.recordEvent(this, 701680);

        try {
            Logging.recordEvent(701681);
        } catch (Throwable var2) {
            Logging.recordEvent(var3, 701682);
            Logging.recordEvent(var2, 701683);
            Logging.recordEvent(var2, 701684);
            throw var2;
        }
    }

    public int Unlogged$Probed$$org$unlogged$demo$controller$testClass$getAdd(int a, int b) {
        byte var3 = 0;

        try {
            Logging.recordEvent(this, 701687);
            Logging.recordEvent(a, 701688);
            Logging.recordEvent(b, 701689);
            Logging.recordEvent(701690);
            Logging.recordEvent(a, 701691);
            Logging.recordEvent(b, 701692);
            int var10000 = a + b;
            Logging.recordEvent(a + b, 701693);
            return var10000;
        } catch (Throwable var4) {
            Logging.recordEvent(var3, 701694);
            Logging.recordEvent(var4, 701695);
            Logging.recordEvent(var4, 701696);
            throw var4;
        }
    }

    public int getAdd(int a, int b) {
        org$unlogged$demo$controller$testClass$unlogged$$mapStore.put("getAdd$(II)I", (Long)org$unlogged$demo$controller$testClass$unlogged$$mapStore.get("getAdd$(II)I") + 1L);
        return Runtime.probeCounter((Long)org$unlogged$demo$controller$testClass$unlogged$$mapStore.get("getAdd$(II)I"), 1L) ? this.Unlogged$Probed$$org$unlogged$demo$controller$testClass$getAdd(a, b) : a + b;
    }

    static {
        Runtime.registerClass("eNqNlNtu00AQhg0tiWgl1F7BE6BUGB92HR9QjYhCLyyqpALU+/Xu2HLl2Mh2W5Wn4GHgpTifz1B20w1dOaHF0sj2fv8/O+NZWdPWM03TrpdVauwXeZmmwAwGk9KgZdFUZZ5DZTRQN8Oc1DVXXvn7YuyRAyK95sxrCq956jVVb2dUVhOS86ee68TUIglmiDiWHYDneZQgvx841GNeHNsBxshNAq7F2yVhUN0R29Fb/73Z7FoTPjMnRWqO4z2gzXT1qjZt/ImrrTx9fiwv6ViXtxl/oWS7wWOZRzcKB/VRQRXZS0V2YUGaV/xlSTK0gL+WXERXchiSPH949AjC+0AhO4BqVDZRkTUZybPHwPSoqJtqnzZZWYTRaHd8b+vBztYwGmzr48MCqrDdvD4iEwg3M57itn4Xahr2NnaVIt4oRVyWRVwTReyQCoom9Czb9Xx9WpLqe6v4RKy1mnsn15dliLyduSzvparDY0Uq1SwfFL66YJePCu+0uMf5pzMmLfjn8yctZF9ak25v81XhF1ttCP7tHP5dlr/opAj+Q3LhvSn9Jx8zUlQ/pWrpTNUvqbqkDGde9VuqugsGI/ixwlfnv8gzTeHKYEQHcnh8/eRUioqnZ3La/vzvpocsJ8bUYUBQHAdW7MR9lyUJ7lOXImJhhPu25eDTgfHUKTQDxkSbvSjaiP6dnPkBwWDZSYIQdRlQnjXxPWrZToD6MfYdlzqey/4AZsAGMw==", "eNpj4NrymoFry3sg/gjEX4D4GxD/AOKfQPyXgWsrAwAIVg5T");
        org$unlogged$demo$controller$testClass$unlogged$$mapStore = new HashMap();
        org$unlogged$demo$controller$testClass$unlogged$$mapStore.put("getAdd$(II)I", 0L);
    }
}
```

3. Decompiled Class (expected)
```java
//
// Source code recreated from a .class file by IntelliJ IDEA
// (powered by FernFlower decompiler)
//

package org.unlogged.demo.controller;

import io.unlogged.Runtime;
import io.unlogged.logging.Logging;
import java.util.HashMap;

public class testClass {

    public testClass() {
        int var1 = 0;
        Logging.recordEvent(701676);
        Logging.recordEvent(701677);
        Logging.recordEvent(701678);
        var1 = 701678;
        super();
        byte var3 = 0;
        Logging.recordEvent(701679);
        Logging.recordEvent(this, 701680);

        try {
            Logging.recordEvent(701681);
        } catch (Throwable var2) {
            Logging.recordEvent(var3, 701682);
            Logging.recordEvent(var2, 701683);
            Logging.recordEvent(var2, 701684);
            throw var2;
        }
    }

    public int Unlogged$Probed$$org$unlogged$demo$controller$testClass$getAdd(int a, int b) {
        byte var3 = 0;

        try {
            Logging.recordEvent(this, 701687);
            Logging.recordEvent(a, 701688);
            Logging.recordEvent(b, 701689);
            Logging.recordEvent(701690);
            Logging.recordEvent(a, 701691);
            Logging.recordEvent(b, 701692);
            int var10000 = a + b;
            Logging.recordEvent(a + b, 701693);
            return var10000;
        } catch (Throwable var4) {
            Logging.recordEvent(var3, 701694);
            Logging.recordEvent(var4, 701695);
            Logging.recordEvent(var4, 701696);
            throw var4;
        }
    }

    public int getAdd(int a, int b) {
        return Runtime.probeCounter("org.unlogged.demo.controller.testClass$getAdd$(II)I", 1L) ? this.Unlogged$Probed$$org$unlogged$demo$controller$testClass$getAdd(a, b) : a + b;
    }

    static {
        Runtime.registerClass("eNqNlNtu00AQhg0tiWgl1F7BE6BUGB92HR9QjYhCLyyqpALU+/Xu2HLl2Mh2W5Wn4GHgpTifz1B20w1dOaHF0sj2fv8/O+NZWdPWM03TrpdVauwXeZmmwAwGk9KgZdFUZZ5DZTRQN8Oc1DVXXvn7YuyRAyK95sxrCq956jVVb2dUVhOS86ee68TUIglmiDiWHYDneZQgvx841GNeHNsBxshNAq7F2yVhUN0R29Fb/73Z7FoTPjMnRWqO4z2gzXT1qjZt/ImrrTx9fiwv6ViXtxl/oWS7wWOZRzcKB/VRQRXZS0V2YUGaV/xlSTK0gL+WXERXchiSPH949AjC+0AhO4BqVDZRkTUZybPHwPSoqJtqnzZZWYTRaHd8b+vBztYwGmzr48MCqrDdvD4iEwg3M57itn4Xahr2NnaVIt4oRVyWRVwTReyQCoom9Czb9Xx9WpLqe6v4RKy1mnsn15dliLyduSzvparDY0Uq1SwfFL66YJePCu+0uMf5pzMmLfjn8yctZF9ak25v81XhF1ttCP7tHP5dlr/opAj+Q3LhvSn9Jx8zUlQ/pWrpTNUvqbqkDGde9VuqugsGI/ixwlfnv8gzTeHKYEQHcnh8/eRUioqnZ3La/vzvpocsJ8bUYUBQHAdW7MR9lyUJ7lOXImJhhPu25eDTgfHUKTQDxkSbvSjaiP6dnPkBwWDZSYIQdRlQnjXxPWrZToD6MfYdlzqey/4AZsAGMw==", "eNpj4NrymoFry3sg/gjEX4D4GxD/AOKfQPyXgWsrAwAIVg5T");
        Runtime.registerMethod("org.unlogged.demo.controller.testClass$getAdd$(II)I")
    }
}
```

- A method is registered in the static fields of the class. 
- probeCounter increments the counter, and decides if the call should go to probed/unprobed method.

**Comment by kartikeytewari-ul** [comment]
Changes:
- move field out of the class
- modify the static method call
- modify the increment and probeCounter method logic
- add required methods in Runtime class

**Comment by kartikeytewari-ul** [comment]
This is fixed with PR #99.
